var/spack/repos/builtin/packages/hpx/package.py

```diff
@@ -163,13 +163,16 @@ class Hpx(CMakePackage, CudaPackage, ROCmPackage):
     patch('git_external.patch', when='@1.3.0 instrumentation=apex')
 
     def instrumentation_args(self):
-        for value in self.variants['instrumentation'].values:
+        args = []
+        for value in self.variants['instrumentation'][0].values:
```

Comment:
This shouldn't be checking the package -- it should define the list in module scope, and use it both to define the variant and here.

Reply:
@becker33 This went in as an hotfix today in #27215 I'll submit a separate PR and ping you.

Reply:
See #27226